src/styles/main.scss

```diff
@@ -0,0 +1,163 @@
+}
+
+.products {
+  width: 100%;
+  height: 100%;
+  overflow: hidden;
+}
+
+.products__wrapper {
+  margin: 0 auto;
+  padding: 50px 40px;
+  display: grid;
+  grid-template-columns: $column-size;
+  grid-template-rows: auto;
+  justify-items: center;
+  justify-content: center;
+  grid-gap: 46px 48px;
+
+  @media (min-width: 1024px) {
+    grid-template-columns: repeat(4, $column-size);
+  }
+
+  @media (min-width: 768px) and (max-width: 1023px) {
+    grid-template-columns: repeat(3, $column-size);
+  }
+
+  @media (min-width: 488) and (max-width: 767) {
+    grid-template-columns: repeat(2, $column-size);
+  }
+}
+
+.card {
+  width: 198px;
+  background-color: white;
+  border: 1px solid $border-color;
+  border-radius: 5px;
+  font-family: $roboto-font;
+  color: $text-color;
+  font-weight: 400;
+  transition: all 300ms linear;
+
+  &:hover {
+    transform: scale(1.2);
+    transition: all 300ms ease;
+  }
+
+  &__container {
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    padding: 16px;
+  }
+
+  &__name {
+    margin-top: 40px;
+  }
+
+  &__title {
+    font-size: $headline-size;
+    line-height: 18px;
+    font-weight: 500;
+    margin: 0;
+    color: $text-color;
+    transition: all 300ms linear;
+  }
+
+  &:hover &__title {
+    color: #34568b;
+    transition: all 300ms linear;
+  }
+
+  &__product-code {
+    margin: 0;
+    font-size: $subscript-size;
+    line-height: 14px;
+    color: $secondary-color;
+    margin-top: 4px;
+  }
+
+  &__review-block {
+    display: flex;
+    justify-content: space-between;
+    margin-top: 16px;
+  }
+
+  &__review-text {
+    font-size: $subscript-size;
+    line-height: 14px;
+    margin: 0;
+  }
+
+  &__price-block {
+    display: flex;
+    justify-content: space-between;
+    margin-top: 24px;
+  }
+
+  &__price-text {
+    font-size: $headline-size;
+    line-height: 18px;
+    color: $secondary-color;
+    margin: 0;
+  }
+
+  &__price {
+    font-size: $price-text;
+    line-height: 18px;
+    font-weight: 700;
+    margin: 0;
+  }
+
+  &__image {
+    width: 100%;
+    height: 134px;
+    padding: 0;
+    margin-top: 16px;
+    position: relative;
+  }
+
+  &__img {
+    display: block;
+    margin: 0 auto;
+    height: 100%;
+  }
+
+  &__button {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    height: 38px;
+    background-color: $accent-color;
+    font-size: $text-size;
+    font-weight: 700;
+    border: 1px solid $accent-color;
+    border-radius: 5px;
+    margin-top: 16px;
+    transition: background-color 300ms linear;
+
+    .button__link {
+      text-align: center;
+      text-decoration: none;
+      color: white;
+      transition: background-color 300ms linear;
+    }
+
+    &:hover {
+      background-color: white;
+      transition: background-color 300ms linear;
+    }
+
+    &:hover .button__link {
+      color: $accent-color;
+      transition: background-color 300ms linear;
+    }
+  }
```

Comment:
Your BEM structure is good, but you can improve it by moving each BEM block to its own file. This will make your styles more modular and easier to maintain. For example, all styles related to `.card` can be moved to `card.scss`.